src/utils/components/ListPageFilter/hooks.ts

```diff
@@ -0,0 +1,87 @@
+import { useCallback, useMemo } from 'react';
+import { useHistory, useLocation } from 'react-router-dom';
+
+type UseRowFiltersParametersType = (inputs: {
+  filterKeys: { [key: string]: string };
+  filters: { [key: string]: string[] };
+}) => [string[], (selected: string[]) => void];
+
+export const useRowFiltersParameters: UseRowFiltersParametersType = ({ filterKeys, filters }) => {
+  const location = useLocation();
+  const { setOrRemoveQueryArgument } = useQueryParamsMethods();
+  const queryParams = useMemo(() => new URLSearchParams(location.search), [location]);
+
+  const selectedRowFilters = useMemo(
+    () =>
+      Object.values(filterKeys)
+        .map((values) => queryParams.get(values)?.split(',') ?? [])
+        .flat(),
+    [queryParams, filterKeys],
+  );
+
+  const syncRowFilterParams = useCallback(
+    (selected: string[]) => {
+      Object.entries(filters).forEach(([key, value]) => {
+        const recognized = selected.filter((item) => value.includes(item));
+        setOrRemoveQueryArgument(filterKeys[key], recognized.join(','));
+      });
+    },
+    [filters, setOrRemoveQueryArgument, filterKeys],
+  );
+
+  return [selectedRowFilters, syncRowFilterParams];
+};
+
+export const useSearchFiltersParameters = () => {
+  const location = useLocation();
+  const queryParams = useMemo(() => new URLSearchParams(location.search), [location]);
+
+  const nameTextFilter = useMemo(() => queryParams.get('name'), [queryParams]);
+  const labelTextFilters = useMemo(
+    () => queryParams.get('labels')?.split(',') ?? [],
+    [queryParams],
+  );
+
+  return { labels: labelTextFilters, name: nameTextFilter };
+};
+
+export const useQueryParamsMethods = () => {
+  const history = useHistory();
+
+  const setAllQueryArguments = (newParams: { [k: string]: string }) => {
+    const params = new URLSearchParams(window.location.search);
+    let update = false;
+
+    Object.entries(newParams).forEach(([k, v]) => {
+      if (params.get(k) !== v) {
+        update = true;
+        params.set(k, v);
+      }
+    });
+
+    if (update) {
+      const url = new URL(window.location.href);
```

Comment:
why simply not use the new ones? why checking?

Reply:
I think it means that if you want to remove something that was already not applied, do not update